proto/viam/app/build/v1/build.proto

```diff
@@ -83,3 +99,79 @@ message ListJobsResponse {
   // jobs is ordered by (build start time, alphabetical platform).
   repeated JobInfo jobs = 1;
 }
+
+message RepoLink {
+  // viam internal ID which maps to an external oauth app
+  string oauth_app_link_id = 1;
+  // OrgID of the module
+  string org_id = 2;
+  // public namespace of the module
+  optional string namespace = 3;
+  // ID of the module (just the name, not the dotted org:module form)
+  string module_id = 4;
+  // git repo in owner/repository form
+  string repo = 5;
+  // email of the viam user who created this
+  string viam_user = 6;
+}
+
+message LinkRepoRequest {
+  RepoLink link = 1;
+}
+
+message LinkRepoResponse {
+  string repo_link_id = 1;
+}
+
+message UnlinkRepoRequest {
+  string repo_link_id = 1;
+}
+
+message UnlinkRepoResponse {}
+
+message ListRepoLinksRequest {}
+
+message ListRepoLinksResponse {
+  repeated RepoLink links = 1;
+}
+
+message ListAppLinksRequest {}
+
+// represents a link between viam users / orgs and an external oauth app
+message AppLink {
+  // viam internal ID which maps to an external oauth app
+  string oauth_app_link_id = 1;
+  // email of the viam user that owns this
+  string viam_user = 2;
+  // username of the external user who created this
+  string external_user = 3;
+  // list of org public namespace (where available) or org UUIDs attached to the external app
+  repeated string org_id_or_ns = 4;
```

Comment:
To support one user attaching multiple orgs to the same app?

Reply:
yes -- main motivation for this was our internal use pattern